File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.ts
function reverseWords(s: string): string {
    s.trim()
    if (s == "") return ""
    const words = s.split(/\s+/);
    let results = "";
    for (let i = words.length-1; i >= 0; i--) {
        results += words[i] + " "
    }
    return results.trim()
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.kt
class Solution {
    fun reverseWords(s: String): String {
        val ss = s.trim()
        if (ss.isEmpty()) return ""

        val sb = StringBuilder()
        val words = ss.split("\\s+".toRegex())

        for (i in words.size - 1 downTo 0) {
            sb.append(words[i])
            if (i > 0) sb.append(" ")
        }

        return sb.toString()
    }
}